### ISSUE TITLE
Usefulness of YouPHPTube-Storage

### ISSUE THREAD

**Issue by jaynelim** [YouPHPTube/YouPHPTube-Storage#2]
hi can u please explain better why this is better than s3 or wasabi? i mean u can get 1TB space on wasabi for $5 per month. i dont know any server seller that will sell you storage space of 1TB for less than $5. so why would buying my own storage to use with this plugin be more cost-effective? i dont get it.

**Comment by akhilleusuggo** [comment]
Cause you're create kind of a close site . The traffic will be delivered from your second machine ( witch is the storage site , as an independent site ) . If you're hosting a big site , with thousands of viewers , you main site will be overloaded , with this storage method , the new videos that you will upload to the storage will be delivered from your second machine , and will not load the main site . And you can set it to multiregional . This plugin will still need a lot of work to get the real benefit , since from my point of view . ( it needs something like google does ( youtube ) , multiregional storage , with the plugin myip , determine the location and redirect the viewer to the closest server to his location . 
Anyway with 1TB that you have , you will be able to host at the least 100 000 mn of streaming ; 10 000 videos each one 10mn at 720p

**Comment by jaynelim** [comment]
@akhilleusuggo still not clear. with wasabi storage, i have unlimited storage. i can store any amount of TB. i can even store 100TB. i just have to pay $5 for each TB. so there is no limit for using this type of storage. and the videos are loaded from wasabi's servers.  wasabi stores my video files and distributes it according to users request. so i have unlimited amount of storage. so why do i need this new plugin? i dont understand still. my real question is 
1. how does this plugin help me when i already have wasabi? 
2. how will this plugin work in the real word situation? i will buy a 20gb vps in uk and connect it to my site using the plugin? then i will buy another 20gb vps in asia to connect to my site using the plugin? then i will buy a 20gb vps in america and add it to my site using this plugin? 
is this how it will work? and the website will redirect automatically users in asia to the asia vps?

**Comment by akhilleusuggo** [comment]
@jaynelim You're wrong , it's 6$ not 5 ( whatever )
1- Your videos stored on Wasabi are getting downloaded to YOUR SERVER , then your server deliver them to your audience . Witch means your server have to work 2 times . Download the video from your storage server to the server , then distribute it .
2- What you're asking on the second question it's exactly what I have told you on the previous point that is missing . 

>  This plugin will still need a lot of work to get the real benefit , since from my point of view . ( it needs something like google does ( youtube ) , multiregional storage , with the plugin myip , determine the location and redirect the viewer to the closest server to his location .

I believe that can be done , but you will need to have on every server the same amount of storage with the same videos , o just detect location and redirect them .

PS: How the plugin works ? The videos stored on xVPS will be delivered from that VPS . If you have 1000 videos on the same VPS , then probably you will be running out of resources. With the plugin you can run multiple VPSs and distribute the works between them .

I'm like you , not convinced with the plugin ( the price if I'm honest ) , but still it's good if they improve it a bit

**Comment by DanielnetoDotCom** [comment]
Hi

I just create this wiki, with a scenario description

https://github.com/YouPHPTube/YouPHPTube-Storage/wiki/Scenario-Description

Please let me know if I miss some information, I will be glad to update it

**Comment by sipero** [comment]
Also rem,ember that will Wasabi if your bandwidth exceeds your file storage then you will not get your bandwidth for free but will need to pay for the bandwidth you use.

I can give you an example of a user that has 33 videos and storage wise these use less than 25GB of storage but over 100TB of monthly bandwidth so if using Wasabi and streaming videos would be paying over $2,000 monthly for bandwidth.

Now what I can do is find a VPS or Dedicated server with unmetered bandwidth at 100mbps which will give me 32TB per month. So this will be much more cost effective even if I need to purchase more than one VPS or Dedicated server.

**Comment by akhilleusuggo** [comment]
Well , most dedicated server nowadays comes with at the least 1gb/s . I'm using GCP and provides 3Gb/s . Where is the usefulness of the 100TB if you can deliver only 100mb/s . ( 100mb/s it's only for 10 users , each one 10mb/s ) . If you are having 100TB , that's mean you're having a million of videos , not even 10gb/s is enough

**Comment by sipero** [comment]
Yes it is true many dedicated servers are now deployed on a 1GBPS port but in most cases the bandwidth will be limited unless you are paying for unmetered 1GBPS bandwidth which gives you around 330TB monthly but on average is going to be $500 per month to upgrade to that level of bandwidth. It is much easier to find servers at reasonable cost with around 33TB of monthly bandwidth so just deploy more servers. 

In my example around 25GB of videos resulted in 100TB of bandwidth in March 2019 with 1 million views in total for those 33 videos. So using the Storage plugin allows me to distribute this and in effect give more redundancy. Often with streaming videos storage required is not substantial but when you have lots of views it is the bandwidth that can be expensive.

**Comment by akhilleusuggo** [comment]
Well , on google cloud I don't have those problems . You can even go for the VM at 14$/month and get unlimited 3gb/s . Probably you don't know some good providers . Good doesn't mean well knows and famous providers . Why do you think providers make a limit to bandwidth ? Do you really think that a provider lose any resources by not limiting the bandwidth ?

**Comment by sipero** [comment]
Can you please share a link so I can check the VM at $14 monthly that you refer to?

**Comment by akhilleusuggo** [comment]
there's no links . You need to create a project , depending where you host it , the size of the disk ( regional or multi regional ) . BTW the multiregional means that your data can be across all the world and ( it's like the plugin ) . 
I can't link , just check google cloud platform ( BTW ! They give you 300$ for first time / first year so you can try )

**Comment by sipero** [comment]
I tried to create a project but with so many options I was not sure what to select to make it something like you described.

**Comment by DanielnetoDotCom** [comment]
@sipero what many options?

**Comment by akhilleusuggo** [comment]
@sipero What do you need exactly ? I will guide you . The first time it's looking so hard , but it's really a game . When you can change any VM parameter any time without any confirmation by the provider and waiting 12h .... this is already a plus! You can get as much as you want dedicated IP ( ofc its gonna cost something )

**Comment by sipero** [comment]
I am interested in knowing the spec of the VPS you are using. I assume I want to choose this option Compute Engine but then wanted a better idea on choosing memory, storage, CPU, and the bandwidth

**Comment by akhilleusuggo** [comment]
The 14-12$ option it's not customized one . You can not change it : 
Machine type>g1-small ( 1vCore , 1.7Gb Ram ) . The core is shared . The stats says it's only 2.0Ghz ( but I don't believe that , it's way more higher clock speed for the performance I have been testing it ) . But I'm not using that one , I gave that one just as example , but I have tested it for YouPHPTube and works perfectly , since you can change the plan when ever you want . I have even tested it on f1-micro ( 1vCore [ uses only 20% of the core ] and 650Mb of ram , and still no lags and ofc 1,5Gb/s .
